iroha/src/asset.rs

```diff
@@ -24,7 +24,16 @@ pub mod isi {
                 .ok_or_else(|| error!("Failed to find asset."))?;
             match world_state_view.asset(&self.destination_id) {
                 Some(asset) => {
-                    asset.quantity += self.object;
+                    if let AssetValue::Quantity(ref mut quantity) = asset.value {
+                        *quantity = quantity
+                            .checked_add(self.object)
+                            .ok_or_else(|| error!("Overflow occured."))?;
+                    } else {
+                        return Err(error!(
+                            "Expected asset with value type: Quantity. Got: {}",
+                            asset.type_string()
+                        ));
+                    }
```

Comment:
Looks like a quite repetitive block.
Maybe we can create a method `AssetValue::try_get<T>(self) -> Result<T, _>` which will return value if it has appropriate type or `Err` with `format!(Expected asset with value type: {}. Got: {}")`?

If it can be a one generic method for different value types we will use it as:
```
*quantity = asset.value.try_get()?;
```